pkg/rhoda/rhoda_test.go

```diff
@@ -18,165 +22,379 @@ import (
+							err := client.Get(context.Background(), k8sClient.ObjectKey{
+								Namespace: namespace,
+								Name:      inventory.Status.Instances[0].Name,
+							}, &testDBaaSConnection)
+							Expect(err).NotTo(HaveOccurred())
+							if len(testDBaaSConnection.Status.Conditions) > 0 {
+								return testDBaaSConnection.Status.Conditions[0].Type == "ReadyForBinding" && testDBaaSConnection.Status.Conditions[0].Status == "True"
+							} else {
+								fmt.Println("testDBaaSConnection.Status.Conditions Len is 0")
+								return false
+							}
+						}, 60*time.Second, 5*time.Second).Should(BeTrue())
+					} else {
+						fmt.Println("No instances to connect")
+					}
+				} else {
+					fmt.Println("Inventory Status is not Ready for connection")
+					Expect(inventory.Status.Conditions[0].Status).To(Equal("True"))
 				}
-				//inventoryData, err := json.MarshalIndent(inventory, "", "  ")
-				//if err != nil {
-				//	fmt.Println("error:", err)
-				//}
-				//fmt.Print(string(inventoryData))
-				//	Eventually(isInventoryReady(c, inventory.ObjectMeta.Name, namespace), 30, 5).Should(BeTrue())
 			})
-		})
+		}
+	})
+
+	It("Console Login UI and pages access check", func() {
+		fmt.Println("Login")
+		ctx, cancel := chromedp.NewContext(context.Background())
+		defer cancel()
+		//Setting timeout
+		//ctx, cancel = context.WithTimeout(ctx, 5*time.Second)
+		//defer cancel()
+
+		var nodes []*cdp.Node
+		//selector for checking the Data Services button
+		selector := "#page-sidebar div ul li button"
+		url := "https://console-openshift-console.apps.rhoda-lab.51ty.p1.openshiftapps.com/dashboards"
+		// url:= "https://console-openshift-console.apps.rhoda-sp-prod.lue0.p1.openshiftapps.com/dashboards"
+		var data string
+
+		if err := chromedp.Run(ctx,
+			SetCookie("openshift-session-token", config.BearerToken, "console-openshift-console.apps.rhoda-lab.51ty.p1.openshiftapps.com", "/", false, false),
+			chromedp.Navigate(url),
+			chromedp.WaitVisible(`#page-sidebar`),
+			chromedp.OuterHTML("html", &data, chromedp.ByQuery),
+			//	chromedp.Nodes(`button`, &nodes, chromedp.ByQueryAll),
+
+			chromedp.Nodes(selector, &nodes),
+			//chromedp.Text(`button`, &selector, chromedp.NodeVisible, chromedp.ByQuery),
+		); err != nil {
+			//panic(err)
+			AbortSuite(err.Error())
+		}
+
+		for _, node := range nodes {
+			//NodeName is the button here, looping through buttons to get Data services
+			fmt.Println(node.NodeName)
+
+			for _, child := range node.Children {
+				//getting Data Services Button
+				fmt.Println(child.NodeValue)
+				if child.NodeValue == "Data Services" {
+					//get the parent's parent which is Li to click on the Database Access button
+					li := child.Parent.Parent
+					//fmt.Println(li)
+					//checkAdminDashboard(li, ctx)
+					var data string
+					selector := "section ul li a"
+					var result []*cdp.Node
+
+					err := chromedp.Run(ctx,
```

Comment:
Clean up this code by having a function that return `li` to you. Make this a Run block right after the previous one, by calling that function for your FromNode()